### PR TITLE
MM-68076 Chunk bulk INSERTs to respect PostgreSQL 65,535 parameter limit

### PR DESCRIPTION
**Pull request by wiggin77**
#### Summary
PostgreSQL's wire protocol uses a 16-bit integer for parameter count, limiting queries to 65,535 parameters. Bulk import of channel members, team members, or thread memberships can exceed this limit, crashing the import.

This PR adds a generic `chunkSlice` helper that splits rows into sub-batches capped at 50,000 parameters, and wraps the INSERT loops in `saveMultipleMembers` (channel, 15 cols), `SaveMultipleMembers` (team, 8 cols), and `SaveMultipleMemberships` (thread, 6 cols).

Normal operations (< 3,333 rows) remain a single INSERT with one bounds-check of overhead. No store interface or caller changes.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-68076

#### Release Note
```release-note
Fixed bulk imports failing on PostgreSQL when channel, team, or thread membership batches exceeded the 65,535 query parameter limit by automatically chunking large INSERT statements.
```

## Change Impact: 🟡 Medium

**Reasoning:** Changes modify critical data persistence logic for bulk membership operations (channels, teams, threads) but are well-isolated to specific store methods with comprehensive test coverage. The chunking logic is straightforward and backward-compatible, though the scope spans multiple core business entities.

**Regression Risk:** Moderate. The modifications touch bulk INSERT execution paths and introduce a new shared utility (`chunkSlice`). However, the chunking logic preserves the original insert semantics—data integrity is not altered, only execution is split. Existing test coverage has been expanded to cover edge cases. The primary risk is incomplete testing of large-scale batch operations or edge cases in the chunking calculation.

**QA Recommendation:** Medium manual QA is recommended. Focus on:
- Large bulk operations (>3,333 rows) to verify chunking behavior
- PostgreSQL-specific testing (the fix targets PostgreSQL parameter limits)
- Concurrent membership operations to ensure no race conditions in chunked inserts
- Data consistency verification after bulk operations
Manual QA can be reduced if continuous integration covers large-scale batch import scenarios.

### CODE REVIEW

**Comment by jwilander** [comment]
@coderabbitai review

File: server/channels/store/sqlstore/thread_store.go [hunk 1]
```diff
@@ -1084,10 +1079,21 @@ func (s *SqlThreadStore) SaveMultipleMemberships(memberships []*model.ThreadMemb
 	}
 	defer finalizeTransactionX(tx, &err)
 
-	_, err = tx.ExecBuilder(query)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to save thread memberships")
+	// 6 columns: PostId, UserId, Following, LastViewed, LastUpdated, UnreadMentions
+	chunks := chunkSlice(memberships, 6)
```
Comment (jwilander) [inline server/channels/store/sqlstore/thread_store.go:1083]: Not blocking for now but we probably don't want this to be hardcoded in the future since it's an easy miss. Maybe create a follow-up ticket?

File: server/channels/store/sqlstore/team_store.go [hunk 2]
```diff
@@ -835,21 +835,22 @@ func (s SqlTeamStore) SaveMultipleMembers(members []*model.TeamMember, maxUsersP
...
+		sql, args, err := query.ToSql()
+		if err != nil {
+			return nil, errors.Wrap(err, "insert_members_to_sql")
+		}
+		if _, err = s.GetMaster().Exec(sql, args...); err != nil {
+			if IsUniqueConstraintError(err, []string{"TeamId", "teammembers_pkey", "PRIMARY"}) {
+				return nil, store.NewErrConflict("TeamMember", err, "")
+			}
+			return nil, errors.Wrap(err, "unable_to_save_team_member")
 		}
```
[the remark this answers is not in the document]
Comment (jwilander) [reply server/channels/store/sqlstore/team_store.go:853]: It's a good call out that we should fix in the future but for our current purposes it's ok. It's rare that chunking will occur at all.

**Review by esarafianou — APPROVED** [review]

File: server/channels/store/sqlstore/channel_store.go [hunk 3]
```diff
@@ -1777,21 +1777,22 @@ func (s SqlChannelStore) saveMultipleMembers(members []*model.ChannelMember) ([]
...
+		sql, args, err := query.ToSql()
+		if err != nil {
+			return nil, errors.Wrap(err, "channel_members_tosql")
+		}
+		if _, err := s.GetMaster().Exec(sql, args...); err != nil {
+			if IsUniqueConstraintError(err, []string{"ChannelId", "channelmembers_pkey", "PRIMARY"}) {
+				return nil, store.NewErrConflict("ChannelMembers", err, "")
+			}
+			return nil, errors.Wrap(err, "channel_members_save")
 		}
```
[the remark this answers is not in the document]
Comment (jwilander) [reply server/channels/store/sqlstore/channel_store.go:1795]: Same comment as below.
